### PR TITLE
Migrate to mdbook and pdoc

### LINKED ISSUE #110: Migrate to pdoc + mdBook
The documentation generated by pdoc (not pdoc3) looks good.
https://pdoc.dev/docs/pdoc.html

### PR DESCRIPTION
**Pull request by yukinarit**
Closes #110 

Hi @alexmisk 

The new documentation consists of [Guide](https://yukinarit.github.io/pyserde/guide/) built with [mdbook](https://github.com/rust-lang/mdBook) and [API docs](https://yukinarit.github.io/pyserde/api/serde.html) built with [pdoc](https://github.com/mitmproxy/pdoc).

I have experimented several documentation systems such as Sphinx, MkDocs, Portray .etc, and concluded using the two best awesome [mdbook](https://github.com/rust-lang/mdBook) and  [pdoc](https://github.com/mitmproxy/pdoc) is the best for pyserde. PTAL!

### CODE REVIEW

**Comment by yukinarit** [comment]
@alexmisk any comment on the changes, especially GitHub Actions script?

**Comment by yukinarit** [comment]
Oops, I was supposed to assign this to you but made a mistake 😄

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,6 +28,9 @@ jobs:
        run: |
          pip install poetry
          poetry install
+         curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain stable
```
Comment (alexmisk) [inline .github/workflows/build.yml:31]: We can use some pre-made GitHub Actions here for setting up [rust toolchain](https://github.com/marketplace/actions/rust-toolchain) or even [mdbook](https://github.com/marketplace/actions/mdbook-action) for better readability. But this line of code do the job and there is no need to modify anything now.

File: docs/book.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[book]
+authors = ["yukinari"]
```
Comment (alexmisk) [inline docs/book.toml:2]: A typo? (yukinari**t**)

**Review by alexmisk — APPROVED** [review]
Looks super cool! I left a few minor comments on build workflow and `docs/book.toml`. Do we need some additions to `setup` command in the Makefile for setting up rust toolchain and building docs with mdbook locally?

File: docs/book.toml (hunk 2 above)
Comment (yukinarit) [reply docs/book.toml:2]: 😄

Comment (yukinarit) [reply docs/book.toml:2]: Seems there are two book.toml. I will delete unused `docs/book.toml`!
* book.toml
* docs/book.toml

Comment (yukinarit) [reply docs/book.toml]: Removed `docs/book.toml`

File: .github/workflows/build.yml (hunk 1 above)
Comment (yukinarit) [reply .github/workflows/build.yml:31]: Good point. But, installing `Rust` + `mdBook` sounds overkill for most of pyserde developers 🤔  I will look for a way to install mdBook as a single binary when I have time.

**Comment by yukinarit** [comment]
@alexmisk 
Thanks for the review! I've addressed for your comments! 
PTAL again, and merge if everything looks good.

**Pull request merged**